data-prepper-plugins/peer-forwarder/src/main/java/com/amazon/dataprepper/plugins/processor/peerforwarder/HashRing.java

```diff
@@ -47,15 +47,18 @@ public HashRing(final PeerListProvider peerListProvider, final int numVirtualNod
         if (hashServerMap.isEmpty()) {
             return Optional.empty();
         }
+
         final byte[] traceIdInBytes = traceId.getBytes();
         final MessageDigest md;
         try {
             md = MessageDigest.getInstance(MD5);
         } catch (NoSuchAlgorithmException e) {
-            throw new RuntimeException(e);
+            throw (AssertionError)new AssertionError("unreachable", e);
```

Comment:
Something I found when figuring out how tests should handle "unreachable" code. Options were to either:
1. MockStatic the MessageDigest `getInstance()` method to throw or
2. Leave it and indicate we should never reach that line of code

Regarding option 2 - I found recommendations for AssertionError to indicate that the assertion of being unreachable was broken. Additionally, because AssertionError is a sublcass of Error, it indicates that the developer _should not_ attempt to catch and handle this as something is fundamentally broken.